Andrii_Voinenko/Lesson4/task1.py

```diff
@@ -0,0 +1,20 @@
+elements = [1, 2, 3, 4, 5, 6, 7, 8]
+odd_elements = []
+odd_index = []
+
+even_elements = []
+even_index = []
+
+for index, element in enumerate(elements):
+    if index % 2 == 0:
+        odd_elements.append(element)
+        odd_index.append(index)
+    else:
+        even_elements.append(element)
+        even_index.append(index)
+
+odd_elements_tuple = list(zip(odd_index, odd_elements))
+even_elements_tuple = list(zip(even_index, even_elements))
```

Comment:
over-engineering.

you can use only two lists odd_elements , even_elements
and in if statements use odd_elements.apend((index, element))